src/components/swap/SwapButton.tsx

```diff
@@ -12,15 +12,26 @@ const StyledSwapButton = styled(ButtonPrimary)<{ gradientColor: string }>`
     !disabled && gradientColor && `linear-gradient(90deg, #2E17F2 19.74%, ${gradientColor} 120.26%)`};
 `
 
+const StyledSwapLoadingButton = styled(ButtonPrimary)`
+  background-image: linear-gradient(90deg, #4c4c76 19.74%, #292942 120.26%);
+  cursor: 'wait';
+`
+
 const StyledPlataformImage = styled.img`
   margin-top: -3px;
   margin-right: 6px;
 `
 
-const StyledSwapButtonText = styled(Text)`
+const StyledSwapButtonText = styled(Text)<{ width?: string }>`
   display: flex;
   height: 16px;
   white-space: pre-wrap;
+  width: ${({ width }) => (width ? `${width}px` : 'auto')};
+`
+
+const StyledLoadingSwapButtonText = styled(StyledSwapButtonText)`
+  justify-content: end;
+  flex: 1.1;
```

Comment:
Did you mean `flex: 1 1`? I don't recall `1.1` being a valid value

Reply:
its like 110% flex can be a positive number 😄 so the other will be 90% or depending on the other elements flex